pkg/nodeconfig/nodeconfig.go

```diff
@@ -0,0 +1,100 @@
+package nodeconfig
+
+import (
+	"encoding/json"
+	"os"
+	"strings"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	NodeArgsAnnotation = "k3s.io/node-args"
+	NodeEnvAnnotation  = "k3s.io/node-env"
+)
+
+func getNodeArgs(nodeName string) (string, error) {
+	nodeArgsList := []string{}
+	for _, arg := range os.Args[1:] {
+		if strings.Contains(arg, "=") {
+			parsedArg := strings.SplitN(arg, "=", 2)
+			nodeArgsList = append(nodeArgsList, parsedArg...)
+			continue
+		}
+		nodeArgsList = append(nodeArgsList, arg)
+	}
+	for i, arg := range nodeArgsList {
+		if isSecret(arg) {
+			nodeArgsList[i+1] = ""
+		}
+	}
+	nodeArgs, err := json.Marshal(nodeArgsList)
+	if err != nil {
+		return "", errors.Wrapf(err, "Failed to retrieve argument list for node %s", nodeName)
+	}
+	return string(nodeArgs), nil
+}
+
+func getNodeEnv(nodeName string) (string, error) {
+	k3sEnv := make(map[string]string)
+	for _, v := range os.Environ() {
+		keyValue := strings.SplitN(v, "=", 2)
+		if strings.HasPrefix(keyValue[0], "K3S_") {
+			k3sEnv[keyValue[0]] = keyValue[1]
+		}
+	}
+	for key := range k3sEnv {
+		if isSecret(key) {
+			k3sEnv[key] = ""
+		}
+	}
+	k3sEnvJSON, err := json.Marshal(k3sEnv)
+	if err != nil {
+		return "", errors.Wrapf(err, "Failed to retrieve environment map for node %s", nodeName)
```

Comment:
Parameterizing the log with the `nodeName` isn't necessary as this should only ever show up in the log on the node in question.